### PR TITLE
Add automatic playlist-ing for team details page

### PR DESCRIPTION
**Pull request by dellagd**
## Description

Modify the team_details page to have an additional button that links to a youtube playlist containing all the match videos of a team for the indicated event.

## Motivation and Context

I found myself having to click a lot to watch a team through the eliminations at their event and thought that was unnecessary. This button lets you sit back and watch a team's performance at an event.

## How Has This Been Tested?

I tested this on events with missing videos, no videos, videos that contain time links (don't work in playlists), and with multiple event entries for a team. 

## Screenshots (if appropriate):
image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

File: templates_jinja2/team_details.html [hunk 1]
```diff
@@ -142,6 +142,8 @@ <h3><a href="/event/{{ comp.event.key_name }}">{{ comp.event.name }}</a></h3>
                   </ul>
                   {% endif %}
                 </p>
+                <a class="btn btn-default" href="https://www.youtube.com/watch_videos?video_ids={{comp.video_ids}}" target="_blank">
```
Comment (andrewda) [inline templates_jinja2/team_details.html]: Only show this if there's at least one video published. Otherwise you're gonna run into a nasty `400` error.

Comment (andrewda) [inline templates_jinja2/team_details.html]: Might also want to add an `&title={{<event name>}}%20videos`

File: renderers/team_renderer.py [hunk 2]
```diff
@@ -104,13 +104,24 @@ def render_team_details(cls, handler, team, year, is_canonical):
...
 
+            video_ids = ""
+            for lv in Match.COMP_LEVELS:
+                ms = matches_organized[lv]
+                for m in ms:
+                    vidstr = ""
+                    for v in m.youtube_videos:
+                        if len(v) > 0:
+                            vidstr = vidstr + v.split("?")[0] + ","
+                    video_ids = video_ids + vidstr
```
Comment (andrewda) [inline renderers/team_renderer.py]: Untested, but you could probably simplify this to something like (also made variables more verbose for readability's sake):

```python
video_ids = []
for level in Match.COMP_LEVELS:
    matches = matches_organized[level]
    for match in matches:
        video_ids += [video.split("?")[0] for video in match.youtube_videos]

# ...

"video_ids": ",".join(video_ids)

# or maybe even better,

"playlist": "https://www.youtube.com/watch_videos?video_ids={}&title={}".format(",".join(video_ids), event.name)
```

Comment (dellagd) [reply renderers/team_renderer.py]: I like the second one, ill implement/test that soon

File: templates_jinja2/team_details.html [hunk 3]
```diff
@@ -142,6 +142,8 @@ <h3><a href="/event/{{ comp.event.key_name }}">{{ comp.event.name }}</a></h3>
...
+                  <span class="glyphicon glyphicon-play-circle"></span> Watch Me at This Event</a>
```
Comment (phil-lopreiato) [inline templates_jinja2/team_details.html]: Copy nit: "Watch All Matches" or something, we don't really use "me" anywhere else on the site like this

Comment (dellagd) [reply templates_jinja2/team_details.html]: Do you think the button location is sufficient to scope "All" to "This this team at this event"? Or should we be more specific with something like "Watch all of ~team number~'s matches"?

Comment (phil-lopreiato) [reply templates_jinja2/team_details.html]: "Watch All" should be sufficient, I think

Comment (dellagd) [reply templates_jinja2/team_details.html]: Sounds fine to me

**Comment by dellagd** [comment]
Updated screenshots

**Comment by fangeugene** [comment]
This looks great. How does it behave with timestamped videos? Does it just link to the beginning of the video?

**Comment by dellagd** [comment]
Yeah, I couldn't think of a more elegant way to handle it that didn't seem to make it a lot more complicated.

**Comment by bdaroz** [comment]
In looking into this there doesn't seem to ge a "good" answer for the start time problem w/o using the Youtube API to make a real playlist.

There also seems to be some anecdotal evidence that there is a 20 video limit to this method. (https://webapps.stackexchange.com/questions/91477/anonymous-youtube-playlist)

**Comment by dellagd** [comment]
Seems to produce playlists of >20 videos fine, see example: https://www.youtube.com/watch_videos?video_ids=sxE3YpGCJvM,rIp6mhagWgk,QDgSMQp4XMk,KviaH0ihmYs,ZeL6dottoY8,fkSKE2TAwnQ,nvhStq7FRT0,kuUTsPrKirk,hA6LsjTog_M,_OZoeiumXig,q-Kx1jA2pXc,ts1IQO5zU64,nCRgbFeIc30,42mqr4UBdfU,rITIL1N9sfU,ojPRy-4ZEMI,2huFkUEiRo4,qtGeAHpz34A,0lwiy_eiUBc,iCYMcCHpfRA,vEMgFcdTk9k,2gAcF3JIh6Y,mxfVvHgpheo,0Lj7u9zqSqU

**Comment by fangeugene** [comment]
@phil-lopreiato I'm fine with this can you approve the changes?

**Comment by phil-lopreiato** [comment]
Oh huh, GitHub keeps the "requested changes" sticky even after the pull was updated. I'm used to the phabricator behavior that resets to "needs review"

**Pull request merged**